server/backend/api/app/crud/crud_user_role.py

```diff
@@ -72,78 +74,118 @@ def create_user_role(
         fam_role.role_type_code == famConstants.RoleType.ROLE_TYPE_ABSTRACT
     )
 
+    create_return_list: List[schemas.FamUserRoleAssignmentCreateResponse] = []
+
     if require_child_role:
         LOGGER.debug(
             f"Role {fam_role.role_name} requires child role "
             "for user/role assignment."
         )
 
-        if (not hasattr(request, "forest_client_number")
-                or request.forest_client_number is None):
+        if (
+            not hasattr(request, "forest_client_numbers")
+            or request.forest_client_numbers is None
+            or len(request.forest_client_numbers) < 1
+        ):
             error_msg = (
-                "Invalid role assignment request. Cannot assign user " +
-                f"{request.user_name} to abstract role {fam_role.role_name}")
+                "Invalid user role assignment request, missing forest client number."
+            )
             raise_http_exception(
                 error_code=famConstants.ERROR_CODE_MISSING_KEY_ATTRIBUTE,
-                error_msg=error_msg
+                error_msg=error_msg,
             )
 
         api_instance_env = crud_utils.use_api_instance_by_app(fam_role.application)
-        validator = UserRoleValidator(request, api_instance_env)
-        if (not validator.forest_client_number_exists()):
-            error_msg = (
-                "Invalid role assignment request. " +
-                f"Forest Client Number {request.forest_client_number} does not exist.")
-            raise_http_exception(
-                error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
-                error_msg=error_msg
+        forest_client_integration_service = ForestClientService(api_instance_env)
+
+        for forest_client_number in request.forest_client_numbers:
+            # validate the forest client number
+            forest_client_validator_return = (
```

Comment:
yeah I agree with you. This one is also copied from admin management, I would prefer to update both if we need to. However, since forest client api now support search multiple forest client numbers in one api call, do you mind if we refactor this when we switch to use that new api?